### PR TITLE
#21 Donations

### PR DESCRIPTION
**Pull request by kushniryb**
Two moments worth mentioning:

1. The usage of interactors allowed us to build reusable donation organizer which can be used both standalone & in conjunction with whatever resource. 
All you have to do in order to use it is to implement your custom organizer which would use  `Donation::Create`, e.g
    ```ruby
    class VacationDonations
      include Interactor

      organize SetupContext, BuyBeer, Donation::Create, SayThankYou
    end
    ```
    The donation form that contains all the required fields for `Donation::Create` organizer can be found here - `shared/donations/_form`, everything else is up to you

2. The absence of tests for `Donation::Create` organizer may seem shocking at first but there's actually no need for that as internally it just delegates everything to the Stripe-related services which are, in fact, tested

:vulcan_salute:

### CODE REVIEW

File: app/interactors/donation/create.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Donation
+  include Charge
+  include Customer
+  include Token
+
+  class Create
+    include Interactor::Organizer
+
+    organize Token::Generate, Customer::Create, Charge::Create
```
Comment (kushniryb) [inline app/interactors/donation/create.rb]: Yes, absolutely
The `Donation::Create` interactor works directly with Stripe so there's nothing persisted to the database at this stage hence no need to rollback, everything else is up to you :)
Quite flexible in my opinion

Comment (skrypalyk) [inline app/interactors/donation/create.rb]: What if in parallel universe you were creating records. I used ActiveRecord::Base.transaction do ... block to rollback. In Interactor case - we should raise ActiveRecord::Rollback exception or what ?

Comment (kushniryb) [inline app/interactors/donation/create.rb]: In such case calling `fail!` method prior to `raise ActiveRecord::Rollback` is sufficient enough

**Comment by sveredyuk** [comment]
@kushniryb rebase with fresh development please

**Comment by kushniryb** [comment]
@sveredyuk it's already rebased

**Comment by sveredyuk** [comment]
@kushniryb Squash and merge with `#21` in commit message. Good job

**Comment by kushniryb** [comment]
@sveredyuk It's done :vulcan_salute:

**Pull request merged**